initializer.go

```diff
@@ -67,6 +91,26 @@ func CreateAzureClientsWithCreds(subscriptionID string, credential azcore.TokenC
 	}, nil
 }
 
+type ClientOptions func(*AzureClientOptions)
+
+func WithAzureClientOptions(options *azcore.ClientOptions) ClientOptions {
+	return func(azureClientOptions *AzureClientOptions) {
+		azureClientOptions.clientOptions = options
+	}
+}
+
+func checkOptionalClientParameters(clientOptions []func(*AzureClientOptions), azureClientOptions *AzureClientOptions) *AzureClientOptions {
+	if clientOptions != nil {
+		for _, optArgs := range clientOptions {
+			optArgs(azureClientOptions)
+			if &azureClientOptions != nil {
+				return azureClientOptions
```

Comment:
all funcs should be applied so that by the time this function retuns all optional params are set in the `AzureClientOptions` struct to be used by the caller

Reply:
@redbaron I'm currently not sure what exactly I should optimize? Should I make the parameter passing more generic (names etc.) or should I use default values inside the struct?

Reply:
@redbaron Could you please explain your idea in more detail?

Reply:
@ShiranAvidov Could please take a look?

Reply:
@redbaron & @8naama Ping :)

Reply:
Hey @ZPascal! :)
The `checkOptionalClientParameters` function returns early, need to ensure it iterates over all `clientOptions` before returning. Here's an updated version:
``` go
if clientOptions != nil {
    for _, optArgs := range clientOptions {
        optArgs(azureClientOptions)
    }
    if azureClientOptions != nil {
        return azureClientOptions
    }
}
return nil
```
We'll incorporate your changes in the next version. Feel free to make this adjustment, and then I'll merge your PR into the new version.

Thank you again for the contribution!

Reply:
Hey @8naama, thank you for the review and the recommendation. I've adapted the code.